### PR TITLE
Java EE Web Profile based application.

### PR DESCRIPTION
**Pull request by violetagg**
This is a test application for https://github.com/cloudfoundry-community/tomee-buildpack/issues/5

### CODE REVIEW

**Comment by cgfrost** [comment]
Hi, this all looks good to me but I'm concerned that it doesn't provide any of the common endpoints that all the other sample applications provide. @nebhale What do you think? 

Common endpoints listed here: https://github.com/cloudfoundry/java-test-applications#output-content

**Comment by violetagg** [comment]
Hi,

I added the common endpoint.
I touched a bit the core module so that I can use it in Java EE environment.
I also added support for FAIL_INIT and FAIL_OOM

Regards,
Violeta

**Comment by violetagg** [comment]
Hi,

Is there anything else that I can do in addition for this change?

Thanks a lot,
Violeta

File: core/src/main/java/com/gopivotal/cloudfoundry/test/core/MemoryUtils.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
  * Utility methods used during memory tests
  */
 @Component
-public final class MemoryUtils {
+public class MemoryUtils {
```
Comment (nebhale) [inline core/src/main/java/com/gopivotal/cloudfoundry/test/core/MemoryUtils.java]: Is the a CDI thing?  That the type can't be `final`?

File: core/src/main/java/com/gopivotal/cloudfoundry/test/core/MemoryUtils.java [hunk 2]
```diff
@@ -63,13 +63,17 @@ public MemoryUtils() {
...
+        outOfMemoryOnStart();
+    }
+
     /**
      * Generates an {@link OutOfMemoryError} if the {@code FAIL_OOM} environment variable is {@code true}. Otherwise
      * does nothing.
      *
      * @return Never returns as it will generate an {@link OutOfMemoryError}
      */
-    @PostConstruct
```
Comment (nebhale) [inline core/src/main/java/com/gopivotal/cloudfoundry/test/core/MemoryUtils.java]: And that it has to return `void`?

**Comment by violetagg** [comment]
Yes that's CDI [1] and it uses annotations defined in JSR 250: Common Annotations for the Java platform. According to this specification [2]:

"
2.5 javax.annotation.PostConstruct
...
- The return type of the method MUST be void
- The method MAY be final or non-final, except in the case of EJBs where it MUST be
non-final.
"

Regards,
Violeta

[1] http://docs.oracle.com/javaee/6/tutorial/doc/gmgkd.html
[2] http://download.oracle.com/otn-pub/jcp/common_annotations-1_2-mrel2-spec/jsr-250-1.2-final.pdf?AuthParam=1456943655_a49f931f0ce762f5042d91f31927484b

**Pull request closed without merging**